### PR TITLE
Provides a DSL for expressing bytes

### PR DESCRIPTION
**Pull request by huntc**
Also ensures that ms and mx are round to the nearest 1k boundary as required by the JVM.

Fixes. #20

### CODE REVIEW

**Comment by huntc** [comment]
This one is for you @hseeberger !

File: README.md [hunk 1]
```diff
@@ -75,12 +75,14 @@ components = {
...
-At a minimum, you will be required to declare the bundle's required number of cpus, its memory and diskspace for your project's build file. For a typical microservice this may look like the following:
+At a minimum, you will be required to declare the bundle's required number of cpus, its memory and disk space for your project's build file. For a typical microservice this may look like the following:
 
 ```scala
+import ByteConversions._
+
 BundleKeys.nrOfCpus := 1.0
-BundleKeys.memory := "64m"
-BundleKeys.diskSpace := "10m"
+BundleKeys.memory := 64.MiB
```
Comment (hseeberger) [inline README.md]: Use postfix for units.

File: src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala [hunk 2]
```diff
@@ -68,6 +68,32 @@ object Import {
     )
   }
 
+  case class Bytes(underlying: Long) extends AnyVal {
+    def round1k: Bytes =
+      Bytes((Math.max(underlying - 1, 0) >> 10 << 10) + 1024)
+  }
+
+  object ByteConversions {
+    implicit class LongToBytes(value: Long) {
```
Comment (hseeberger) [inline src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala]: Using a `Long` here will open the door for overflows for the below methods. Maybe use an `Int`?

Comment (hseeberger) [inline src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala]: Naming should be `FooOps`, i.e. `LongOps` or `IntOps` (overflow).

File: src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala [hunk 3]
```diff
@@ -95,8 +121,8 @@ object SbtBundle extends AutoPlugin {
     bundleType := Universal,
     startCommand := Seq(
       (file((packageName in Universal).value) / "bin" / (executableScriptName in Universal).value).getPath,
-      s"-J-Xms${uomToBytes(memory.value)}",
-      s"-J-Xmx${uomToBytes(memory.value)}"
+      s"-J-Xms${memory.value.round1k.underlying}",
```
Comment (hseeberger) [inline src/main/scala/com/typesafe/sbt/bundle/SbtBundle.scala]: Nice!

File: src/sbt-test/sbt-bundle/basic/build.sbt [hunk 4]
```diff
@@ -25,7 +25,7 @@ checkBundleConf := {
                             |system     = "simple-test-0.1.0-SNAPSHOT"
                             |nrOfCpus   = 1.0
                             |memory     = 67108864
-                            |diskSpace  = 10485760
+                            |diskSpace  = 10000000
```
Comment (hseeberger) [inline src/sbt-test/sbt-bundle/basic/build.sbt]: Use this nice DSL here, too?

**Comment by hseeberger** [comment]
There should be some doco wrt MB vs MiB.

File: src/sbt-test/sbt-bundle/basic/build.sbt (hunk 4 above)
Comment (huntc) [inline src/sbt-test/sbt-bundle/basic/build.sbt]: I'm actually wanting to check that the bytes are correctly converted here.

File: README.md (hunk 1 above)
Comment (huntc) [inline README.md]: ...but postfix can only be used with functions... how would you do that...?

Comment (huntc) [inline README.md]: The suffix notation with arity-0 methods is discouraged:

http://docs.scala-lang.org/style/method-invocation.html#suffix_notation

Comment (hseeberger) [inline README.md]: WAT?

``` scala
scala> implicit class IntOps(n: Int) {
     |   def k: Long = n*1000L
     | }
defined class IntOps
warning: previously defined object IntOps is not a companion to class IntOps.
Companions must be defined together; you may wish to use :paste mode for this.

scala> 1.k
res8: Long = 1000

scala> 1 k
res9: Long = 1000
```

Comment (hseeberger) [inline README.md]: Yes of course, for normal code, but DSLs are different, many things are allowed there.

Comment (hseeberger) [inline README.md]: In our config we always use `3 seconds` instead of `3.seconds`.

Comment (huntc) [inline README.md]: Er - no we don't. :-)

Besides, removing the `.` doesn't work.

**Comment by huntc** [comment]
Included doco re. MiB.

File: README.md [hunk 5]
```diff
@@ -75,15 +75,19 @@ components = {
...
+BundleKeys.diskSpace := 10.MB
 BundleKeys.roles := Set("web-server")
 ```
 
+(You'll note that the international standards for supporting [binary prefixes](http://en.wikipedia.org/wiki/Binary_prefix), as in `MiB`, is supported).
```
Comment (hseeberger) [inline README.md]: Nice! I learned something new ;-)

**Pull request merged**